### PR TITLE
fix: add cron validation to jobVerifier

### PR DESCRIPTION
**Pull request by Hayden-Yu**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                       |

Related issues: #454 
Closes

Further  information:

### CODE REVIEW

**Comment by enudler** [comment]
hi @Hayden-Yu 
It's great!

What do you think about preventing common chars of cron expression that are not supported in the UI form?

File: ui/src/features/components/JobForm/validator.js [hunk 1]
```diff
@@ -81,12 +81,8 @@ function rampToValidator (value, allState) {
 }
 
 function cronValidator (value, allState) {
-  if (value) {
-    try {
-      cronstrue.toString(value)
-    } catch (err) {
-      return 'illegal cron input'
-    }
+  if (value && !cron.isValidCron(value, { alias: true, seconds: true })) {
```
[the remark this answers is not in the document]
Comment (Hayden-Yu) [reply ui/src/features/components/JobForm/validator.js]: Oops.. Sorry about that! It should work this time.

Also while I was testing I found that in cron expressions like this one: '0 20 0 7 OCT ? 2020', people put year 2020 in the end to restrict the job to run only during year 2020. However, this is not a [standard cron pattern](http://crontab.org/). It's not supported by the version of cron package we use on the backend but that package does not throw error on it either. The '2020' will be silently ignored by the backend, which is likely not expected by our user.

It looks like the cron package [recently added validation on this as well](https://github.com/kelektiv/node-cron/blob/1e047913405de9eb70eb1c01bdacf63b8c90d0d3/lib/cron.js#L420) so we should be able to solve this by bumping up the version of the cron package. Should I go ahead and do so?

[the remark this answers is not in the document]
Comment (enudler) [reply ui/src/features/components/JobForm/validator.js]: Yes please @Hayden-Yu

[the remark this answers is not in the document]
Comment (manorlh) [reply ui/src/features/components/JobForm/validator.js]: good fixed.
please change 
```
 {oneItem.name === 'cron_expression' && <CronViewer value={cron_expression} />}
```
to  
```
{oneItem.name === 'cron_expression' && !this.state.errors['cron_expression'] && <CronViewer value={cron_expression} />}
see what happen with  input: 5 5 5 5 10

thanks


```

[the remark this answers is not in the document]
Comment (Hayden-Yu) [reply ui/src/features/components/JobForm/validator.js]: @manorlh 
Good point! Thank you!

**Comment by enudler** [comment]
Thanks @Hayden-Yu
Great work! 
Much appreciated!

**Pull request merged**